scripts/uosc.lua

```diff
@@ -4322,8 +4322,8 @@ mp.observe_property('demuxer-cache-state', 'native', function(prop, cache_state)
 		}
 	end
 	table.sort(ranges, function(a, b) return a[1] < b[1] end)
-	if cache_state['bof-cached'] then ranges[1][1] = 0 end
-	if cache_state['eof-cached'] then ranges[#ranges][2] = state.duration end
+	if cache_state and cache_state['bof-cached'] then ranges[1][1] = 0 end
+	if cache_state and cache_state['eof-cached'] then ranges[#ranges][2] = state.duration end
```

Comment:
Wrap it in a single `if cache_state then`.

Reply:
At the beginning of the function is also a nil check for cache_state. I could initialize everthing at the beginning in a single if.

Reply:
Is this good?